### DISCUSSION TITLE
Multi-GPUs DDP - How the dataset is distributed accross the GPUs

### DISCUSSION THREAD

**Discussion by KevinCrp** [Lightning-AI/pytorch-lightning#13342] [DDP / multi-GPU / multi-node] [answered]
Hi,

I'm using some GPUs and the Distributed-Data-Parallel strategy and I want to know how the global dataset is split across all GPUs.

Is it split iteratively: 
  * The N first graphs go-to GPU 0
  * The N next go-to GPU 1
  * ...
  *  Until the end ?

Is it randomly distributed?

Is it another way?

I looked at the DistributedSampler class, but I didn't find the answer.

**Comment by akihironitta** [comment, marked as answer]
I believe [this line](https://github.com/pytorch/pytorch/blob/63c0bcf887736810dd2ca6e8b671439c67e58ed6/torch/utils/data/distributed.py#L118) in PyTorch code explains it all:
```python
indices = indices[self.rank:self.total_size:self.num_replicas]
```
